.github/workflows/task-list-checker.yml

```diff
@@ -0,0 +1,20 @@
+name: GitHub Task List Checker
+on:
+  pull_request:
+    branches:
+      - master
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - edited
+jobs:
+  task-list-checker:
+    runs-on: ubuntu-latest
+    permissions:
+      statuses: write  # required for https://docs.github.com/rest/commits/statuses#create-a-commit-status

```

Comment:
permissions should be restricted as much as possible for all other workflows as well

also this can be defined at workflow scope which may be more convenient https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

Reply:
also I don't think your comment is correct. this is good practice of course but shouldn't be required with a permissive token https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

Reply:
the comment is a leftover from past attempts to make it work